cmd/ingress-operator/start.go

```diff
@@ -134,6 +134,9 @@ func createDNSProvider(cl client.Client, operatorConfig operatorconfig.Config, d
 
 	switch platformStatus.Type {
 	case configv1.AWSPlatformType:
+		if len(platformStatus.AWS.Region) == 0 {
+			return nil, fmt.Errorf("region is required")
+		}
```

Comment:
I wonder if this is backwards-compatible with existing storage (https://github.com/openshift/api/commit/dedfb47b1ff14bbd03727fa5720f63d8dc7bea5b)...

See https://github.com/openshift/cluster-ingress-operator/blob/master/cmd/ingress-operator/start.go#L194.

Is this new check necessary?

Reply:
Nevermind, I misread — by the time you get here, the platformStatus has been created by getPlatformStatus and should be normalized/valid no matter the source. I think your validation here is correct.

Reply:
@ironcladlou it's not a new check. I bubbled [this check](https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/dns/aws/dns.go#L101-L103) up from the dns/aws pkg.